src/software/ai/navigator/obstacle/obstacle_test.cpp

```diff
@@ -3,231 +3,120 @@
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    EXPECT_EQ(expected.getPoints(), polygon_obstacle.getPolygon().getPoints());
+    std::ostringstream polygon_ss;
+    polygon_ss << expected;
+    EXPECT_TRUE(polygon_obstacle.toString().find(polygon_ss.str()) != std::string::npos);
 }
 
-// obstacle with robot additional radius buffer = 0, additional velocity projection buffer
-// = 0 centred at (1,2) and with (3,2) velocity
-TEST(NavigatorObstacleTest, shifted_no_buffer_velocity_obstacle_polygon)
+TEST(NavigatorObstacleTest, create_from_circle)
 {
-    double TEST_EPSILON    = 1e-3;
-    Timestamp current_time = Timestamp::fromSeconds(123);
-    Robot robot            = Robot(3, Point(1, 2), Vector(3, 2), Angle::fromRadians(2.2),
-                        AngularVelocity::fromRadians(-0.6), current_time);
-    auto p_pointer = Obstacle::createRobotObstacleWithBufferParams(robot, true, 0.0, 0.0)
-                         .getBoundaryPolygon();
-
-    EXPECT_TRUE(p_pointer);
-
-    if (p_pointer)
-    {
-        Polygon pg = *p_pointer;
-        // visually verified
-        Point pt0 = Point(0.885, 2.173);
-        Point pt1 = Point(.793, 1.987);
-        Point pt2 = Point(.908, 1.814);
-        Point pt3 = Point(1.115, 1.827);
-        Point pt4 = Point(4.115, 3.827);
-        Point pt5 = Point(3.885, 4.173);
-        EXPECT_DOUBLE_EQ(pg.getPoints().size(), 6);
-        EXPECT_NEAR(pg.getPoints()[0].x(), pt0.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[0].y(), pt0.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].x(), pt1.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].y(), pt1.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].x(), pt2.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].y(), pt2.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    Circle expected({2, 2}, 3);
+    CircleObstacle circle_obstacle(expected);
+
+    EXPECT_EQ(circle_obstacle.getCircle(), expected);
+    std::ostringstream circle_ss;
+    circle_ss << expected;
+    EXPECT_TRUE(circle_obstacle.toString().find(circle_ss.str()) != std::string::npos);
 }
 
-// obstacle with robot additional radius buffer = 0.4, additional velocity projection
-// buffer = 1.8 centred at (-1,-2) and with (-3,2) velocity
-TEST(NavigatorObstacleTest, shifted_with_buffer_velocity_obstacle_polygon)
+TEST(NavigatorObstacleTest, rectangle_obstacle_util_functions)
```

Comment:
This test is testing several functions at once. It should be broken up to test each function separately.

Reply:
broke it up